sdk/src/main/java/dev/cdevents/CDEvents.java

```diff
@@ -116,35 +110,119 @@ public static CDEvent cdEventFromJson(String cdEventJson) {
     }
 
     /**
-     * Validates the cdEventJson against the Schema URL.
-     * @param cdEventJson
-     * @return true, If cdEventJson is valid
+     * Creates a CloudEvent from the custom cdEvent.
+     * @param <T> customCDEvent class
+     * @param customCDEvent custom CDEvent class object of type <T>
+     * @param validateContextSchema true If validation needed against context.schemaUri
+     * @return CloudEvent
      */
-    public static boolean validateCDEventJson(String cdEventJson) {
-        String eventType = getUnVersionedEventTypeFromJson(cdEventJson);
-        CDEventConstants.CDEventTypes cdEventType = getCDEventTypeEnum(eventType);
+    public static <T extends CDEvent> CloudEvent customCDEventAsCloudEvent(T customCDEvent, boolean validateContextSchema) {
+        Optional.of(customCDEvent)
```

Comment:
I'm not so sure about the use of `Optional` here instead of a null and validation check

Reply:
yeah make sense